### PR TITLE
Set DYLD_LIBRARY_PATH to test with the just-built stdlib

### PR DESCRIPTION
**Pull request by JDevlieghere**
With Swift in the OS, we need to ensure that the test suite runs with
the standard library we just built, instead of the one form the
operating system. We do this by telling dyld to pick the one from our
build directory by setting the `DYLD_LIBRARY_PATH`. Swift takes the same
approach in their test suite.

### CODE REVIEW

**Comment by JDevlieghere** [comment]
@swift-ci please test

**Comment by adrian-prantl** [comment]
Does this not work at all at the moment, or only for some tests?
I wonder if it would be better to put this into the lit/lit-lldb-init file we are passing to every lldb instance.

**Comment by JDevlieghere** [comment]
> I wonder if it would be better to put this into the lit/lit-lldb-init file we are passing to every lldb instance.

That's exactly what this patch does :-) Maybe I'm misunderstanding the question?

**Review by adrian-prantl — APPROVED** [review]
Sorry, I see it now!

**Review by dcci — APPROVED** [review]
This is correct. I'll let you click the merge button.

**Pull request merged**
